src/PdoConnection.php

```diff
@@ -58,6 +58,16 @@ class PdoConnection
     private $reuseConnection = true;
 
     /**
+     * @var int Number of open transactions
+     */
+    private $transactionCount = 0;
+
+    /**
+     * @var bool Flag that tells whether a rollback is needed
+     */
+    private $rollbackNeeded = false;
```

Comment:
Performing a rollback within a commit is not a good design choice, imo. If you can't commit, or you can't rollback you probably need to throw an exception because that's definitely programmer error.

Also, if they is just a queue system, why not use a queue?